modyn/supervisor/internal/pipeline_executor/pipeline_executor.py

```diff
@@ -124,6 +130,65 @@ def _persist_pipeline_log(self) -> None:
             json.dump(self.pipeline_log, logfile, indent=4)
 
+    # pylint: disable=too-many-locals
+    def _start_evaluations(
+        self, trigger_id: int, model_id: int, trigger_set_first_timestamp: int, trigger_set_last_timestamp: int
+    ) -> None:
+        assert self.grpc.evaluator is not None, "Evaluator not initialized."
+        eval_dataset_config = self.pipeline_config["evaluation"]["dataset"]
+        eval_dataset_id = eval_dataset_config["dataset_id"]
+        eval_strategy_config = self.pipeline_config["evaluation"]["eval_strategy"]
+        eval_strategy_module = dynamic_module_import("modyn.supervisor.internal.eval_strategies")
+        eval_strategy: AbstractEvalStrategy = getattr(eval_strategy_module, eval_strategy_config["name"])(
+            eval_strategy_config["config"]
+        )
+        if EVALUATION_RESULTS not in self.pipeline_log:
+            self.pipeline_log[EVALUATION_RESULTS] = {}
+
+        self.pipeline_log[EVALUATION_RESULTS][trigger_id] = {}
+
+        for previous_split, current_split in eval_strategy.get_eval_intervals(
+            first_timestamp=trigger_set_first_timestamp,
+            last_timestamp=trigger_set_last_timestamp,
+        ):
+            logger.info(f"Evaluation Starts for model {model_id} on split {previous_split} to {current_split}.")
+            request = GRPCHandler.prepare_evaluation_request(
+                eval_dataset_config,
+                model_id,
+                self.pipeline_config["evaluation"]["device"],
+                previous_split,
+                current_split,
+            )
+            self._sw.start("evaluate_model", overwrite=True)
+            response: EvaluateModelResponse = self.grpc.evaluator.evaluate_model(request)
+            interval_name = f"{previous_split}-{current_split}"
+            if not response.evaluation_started:
+                reason = EvaluationAbortedReason.DESCRIPTOR.values_by_number[response.eval_aborted_reason].name
+                logger.error(
+                    f"Evaluation for model {model_id} on split {previous_split} to {current_split} not started with "
+                    f"reason: {reason}."
+                )
+                self.pipeline_log[EVALUATION_RESULTS][trigger_id][interval_name] = {
+                    "failure_reason": reason,
+                }
+                self._sw.stop("evaluate_model")
+            else:
+                logger.info(f"Evaluation started for model {model_id} on split {previous_split} to {current_split}.")
+                reporter = EvaluationStatusReporter(
+                    self.eval_status_queue, response.evaluation_id, eval_dataset_id, response.dataset_size
+                )
+                reporter.create_tracker()
+                evaluation = {response.evaluation_id: reporter}
+                assert self.current_training_id is not None, "Training ID not set."
+                self.grpc.wait_for_evaluation_completion(self.current_training_id, evaluation)
+
```

Comment:
This json writer is actually not needed as we store the results in the pipeline log rather than in a separate eval file. We keep it for the moment to re-use its code to issue the `EvaluationResultRequest` and parse the `EvaluationResultReponse`.

We should have a further PR to clean up the logic here.